src/client.js

```diff
@@ -3991,14 +4008,25 @@ MatrixClient.prototype.getCanResetTimelineCallback = function() {
  */
 MatrixClient.prototype.relations =
 async function(roomId, eventId, relationType, eventType, opts = {}) {
+    const isEncrypted = this.isRoomEncrypted(roomId);
+    const fetchedEventType = _getEncryptedIfNeededEventType(this, roomId, eventType);
+
     const result = await this.fetchRelations(
         roomId,
         eventId,
         relationType,
-        eventType,
+        fetchedEventType,
         opts);
+
+    let events = result.chunk.map(this.getEventMapper());
+    if (isEncrypted) {
+        await Promise.all(events.map(e => {
+            return new Promise(resolve => e.once("Event.decrypted", resolve));
```

Comment:
Reactions in an encrypted room won't be encrypted events themselves, so we can't use `isEncrypted` for the whole room to decide whether to await `Event.decrypted`.

Since the event mapper will invoke `attemptDecryption`, I think you only want to wait for `Event.decrypted` if `event.isBeingDecrypted()`.